pkg/collectors/pod.go

```diff
@@ -145,6 +144,18 @@ var (
 		[]string{"namespace", "pod", "container"}, nil,
 	)
 
+	descPodContainerResourceRequests = prometheus.NewDesc(
+		"kube_pod_container_resource_requests",
+		"The number of requested request resource by a container.",
+		[]string{"namespace", "pod", "container", "node", "resource", "unit"}, nil,
+	)
+
+	descPodContainerResourceLimits = prometheus.NewDesc(
+		"kube_pod_container_resource_limits",
+		"The number of requested limit resource by a container.",
+		[]string{"namespace", "pod", "container", "node", "resource", "unit"}, nil,
```

Comment:
I'm not sure the `unit` label will be confusing for use with device plugins as we can't tell the unit. @derekwaynecarr what's your thought on that?

Reply:
@brancz That's right. I have also think about this when prepare this PR but having not found any better names/values.

Reply:
/c @derekwaynecarr @brancz

Reply:
I talked to @derekwaynecarr and devices have no unit and are always integers, so I think you can decide on either. I can see the unit label being valuable to see that something is indeed a device plugin, but also cpu, memory and storage seems pretty standard so it may not be all that necessary. Up to you either works for me.

Reply:
I suggest we add a `unit` label to help end users learn the unit of the value. I.e., it should be clear for users to learn that memory related limit and request are in `byte` instead of `GB` or `KB`. :)

Reply:
for reference:
https://github.com/kubernetes/kubernetes/blob/8f20a815e5ef27d4d247e42bff90e42706397ca2/pkg/apis/core/helper/helpers.go#L263

validates that any extended resource is integral.

Reply:
ephemeral storage is bytes as well.

Reply:
> cpu is millicores

The unit for cpu limit and resource has been converted from millicores to cores by devided by 1000. So maybe cpu is cores. WDYT?

@derekwaynecarr

Reply:
> all extended resources (device plugin or otherwise in a qualified namespace) are integral

IIUC, you mean that we should
* check whether a resource name is an extended resource
* unit for extended resource should be integer as it is must be an integer

Reply:
kube-state-metrics should not need to check whether the value is integer as this has been done in api-server.

@derekwaynecarr